commandeer.nim

```diff
@@ -284,6 +284,26 @@ template subcommand*(identifier: untyped, subcommandName: string,
     statements
     currentMapping = subCommands[""]
 
+template subcommand*(identifier: untyped, subcommandNames: openArray[string],
```

Comment:
Something that will go hand in hand with your remarks and @moigagoo 's is to replace `openArray` by `varargs`. This way the developer can do:

```nim
commandline:
  subcommand build, ["build", "make"]:  # this
    option projectPath, string, "path", "p", "."
    option targetFormat, string, "target", "t"
    exitoption "help", "h", buildUsage
  subcommand add, "add", "a":  # or that
    argument added, string
```
at her discretion!

In an ideal world there would be only one `subcommand` template, but templates and `varargs` are buggy/broken in Nim right now (see https://github.com/nim-lang/Nim/issues/2459, https://github.com/nim-lang/Nim/issues/1862 and others). So we have to go for 2 templates for now :( .

Reply:
I'm not sure we need two templates - I've tried it now, replacing `openArray` with `varargs`, and it seems to work with both of the styles of names/aliases you show, along with just a single name. 

In the issues I've seen it seems the errors happen when the template has the `{.immediate.}` pragma, or the `varargs` is empty, neither of which should happen in this case -